lib/SftpClient.js

```diff
@@ -1,24 +1,31 @@
+const fs = require('fs-extra')
 const SFTP = require('sftp-promises')
 const FileParser = require('ftp/lib/parser')
 const { PassThrough } = require('stream')
 
 class SftpClient {
-  constructor ({ host, port = 21, user, password }) {
+  constructor ({ host, port = 22, user, password, privateKey, passphrase }) {
     this.host = host
     this.port = port
     this.user = user
     this.password = password
+    this.privateKey = privateKey
+    this.passphrase = passphrase
     this.client = new SFTP()
     this.session = null
   }
 
   async connect () {
-    this.session = await this.client.session({
+    const options = {
       host: this.host,
       port: this.port,
       username: this.user,
-      password: this.password
-    })
+      password: this.password,
+      privateKey: this.privateKey && await fs.readFile(this.privateKey),
```

Comment:
I think this should be a string and the caller has to take care of how to get the key string. This is useful for cases where the pipeline runs in a GitLab instance and the key is provides as env variable. Reading from the file system is still possible, but right now a little bit complicated: 

```
"${require('fs').readFileSync('/full/path/myKeyFile.key').toString()}"^^code:EcmaScriptTemplateLiteral
```
We should add a loader to read the content of files on the local file system via datatype.

Reply:
I agree getting this from ENV is more likely in a pipeline. I do that already in my case.

Reply:
Ok, will change